### ISSUE TITLE
[stable/mongodb] Not working without password

### ISSUE THREAD

**Issue by shield-9** [helm/charts#16450]
**Describe the bug**
When `usePassword` is set to `false`, mongodb-primary-0 fails to start.

**Version of Helm and Kubernetes**:
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.2", GitCommit:"f6278300bebbb750328ac16ee6dd3aa7d3549568", GitTreeState:"clean", BuildDate:"2019-08-05T09:23:26Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.5-ca", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"dirty", BuildDate:"2019-03-29T02:08:05Z", GoVersion:"go1.11.1", Compiler:"gc", Platform:"linux/amd64"}
$ helm version
Client: &version.Version{SemVer:"v2.14.3", GitCommit:"0e7f3b6637f7af8fcfddb3d2941fcc7cbebb0085", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.14.0", GitCommit:"05811b84a3f93603dd6c2fcfe57944dfa7ab7fd0", GitTreeState:"clean"}
```

**Which chart**:
stable/mongodb (5.20.3 or later)

**What happened**:
```
$ kubectl logs demo-mongodb-primary-0 mongodb-primary
Welcome to the Bitnami mongodb container
Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mongodb
Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mongodb/issues
Send us your feedback at containers@bitnami.com
INFO  ==> ** Starting MongoDB setup **
INFO  ==> Validating settings in MONGODB_* env vars...
ERROR ==> MONGODB_ROOT_PASSWORD have to be set on a 'primary' node!
```

**What you expected to happen**:
When `usePassword` is set to `false`, mongodb-primary-0 start without problem.

**How to reproduce it** (as minimally and precisely as possible):
My values.yaml is:
```yaml:values.yaml
replicaSet:
  enabled: true
metrics:
  enabled: true
livenessProbe:
  enbaled: true
readinessProbe:
  enabled: true
usePassword: false
```

**Comment by miguelaeh** [comment]
Hi @shield-9 ,
I tried to reproduce the issue but in my case I cannot get that error. I installed the Chart using:
```
$ helm install stable/mongodb -n mongodb --set usePassword=false
```
Also I tried with a normal deployment and both work as expected.
Could you try that command? If it doesn't work, could you try to clean your environment? Such as PVC and so on that are not removed with the `--purge` option on the `helm delete`.
Also, I see your `values.yaml` has a typo:
```
livenessProbe:
  enbaled: true <-- this should be enabled: true
```

**Comment by shield-9** [comment]
Could you try this?
`helm install stable/mongodb -n mongodb --set usePassword=false --set replicaSet.enabled=true`

> Also, I see your values.yaml has a typo:
Oops, thank you! It is my mistake while creating example YAML.

**Comment by alec-benson** [comment]
> Hi @shield-9 ,
> I tried to reproduce the issue but in my case I cannot get that error. I installed the Chart using:
> 
> ```
> $ helm install stable/mongodb -n mongodb --set usePassword=false
> ```
> 
> Also I tried with a normal deployment and both work as expected.
> Could you try that command? If it doesn't work, could you try to clean your environment and execute the command again? Remove things such as PVC and so on that are not removed with the `--purge` option on the `helm delete`.
> Also, I see your `values.yaml` has a typo:
> 
> ```
> livenessProbe:
>   enbaled: true <-- this should be enabled: true
> ```

I am seeing the same problem. Using the helm command you suggested the primary fails to spin up and I see `MONGODB_ROOT_PASSWORD have to be set on a 'primary' node!` in the logs

**Comment by shield-9** [comment]
@davide-talesco I'm using modified container image to avoid this issue.
https://gist.github.com/shield-9/4ae1abd55efb9314b554b2692df948de

**Comment by jwalton** [comment]
One workaround is to set `replicaSet.enabled=false` (which is obviously a slightly lame workaround, because you don't have a replica set anymore).

The other is, obviously to set `usePassword: true` (which is obviously also a lame workaround).

**Comment by jwalton** [comment]
I've found another awful way around this.  :P  It seems the issue isn't actually with the helm chart, but with the bitnami mongodb image.  If you use older versions of the mongodb image, it'll fix this problem for you:

```yaml
image:
  tag: 3.2.11-r2
replicaSet:
  enabled: true
  name: rs0
  replicas:
    secondary: 1
    arbiter: 1
usePassword: false
# Need this because the older bitnami mongodb images won't work if you're not running as root
securityContext:
  enabled: false
```

This comes up.  In fact, if you set `usePassword: true` here, this will fail with:

```sh
Error executing 'postInstallation': Not possible to configure replica set scenario using authentication. Please do not provided any of the following properties: username, password, database, rootPassword.
```

-_-

**Comment by MihaiAnei** [comment]
I think this might need reopening considering the details in the linked issue :)

**Comment by arizawan** [comment]
I tried this command after many attempts. It seems like working for me. 

`helm -n oih-dev-ns install mongodb bitnami/mongodb --set usePassword=false --set replicaSet.enabled=false --set persistence.enabled=true --set mongodbRootPassword="" --set mongodbPassword=""`

I am able to login without password

**Comment by humblewolf** [comment]
Just for info:: Fix for this issue was first shipped in mongo helm chart version 7.10.9 https://github.com/bitnami/charts/commit/ff1db9d4264e63ff3afb340ddaf0a4ff89ff8a82#diff-36d6c73e8bcce8b9eb984fba150cacb36170c85f85f688d273bff9791b47d08a
